pennylane/devices/preprocess.py

```diff
@@ -136,46 +139,57 @@ def validate_device_wires(
             f"Abstract wires are not yet supported."
         )
 
-    if wires:
+    if not wires:
+        return (tape,), null_postprocessing
 
-        if any(qml.math.is_abstract(w) for w in wires):
-            raise WireError(
-                f"Cannot run circuit(s) on {name} as abstract wires are present in the device: {wires}. "
-                f"Abstract wires are not yet supported."
-            )
+    if any(qml.math.is_abstract(w) for w in wires):
+        raise WireError(
+            f"Cannot run circuit(s) on {name} as abstract wires are present in the device: {wires}. "
+            f"Abstract wires are not yet supported."
+        )
 
-        if extra_wires := set(tape.wires) - set(wires):
-            raise WireError(
-                f"Cannot run circuit(s) on {name} as they contain wires "
-                f"not found on the device: {extra_wires}"
-            )
+    if extra_wires := tuple(
+        w for w in set(tape.wires) - set(wires) if not isinstance(w, DynamicWire)
+    ):
+        raise WireError(
+            f"Cannot run circuit(s) on {name} as they contain wires "
+            f"not found on the device: {extra_wires}"
+        )
 
-        modified = False
-        new_ops = None
-        for i, op in enumerate(tape.operations):
-            if isinstance(op, qml.Snapshot):
-                mp = op.hyperparameters["measurement"]
-                if not mp.wires:
-                    if not new_ops:
-                        new_ops = list(tape.operations)
-                    modified = True
-                    new_mp = copy(mp)
-                    new_mp._wires = wires  # pylint:disable=protected-access
-                    new_ops[i] = qml.Snapshot(
-                        measurement=new_mp, tag=op.tag, shots=op.hyperparameters["shots"]
-                    )
-        if not new_ops:
-            new_ops = tape.operations  # no copy in this case
+    if any(isinstance(w, DynamicWire) for w in tape.wires) and any(
```

Comment:
This function is called after resolving dynamic wires, in the stack, right?
Would this clause ever be triggered? 
Or are we only adding it to support other preprocessing pipelines that might not have called the `resolve_dynamic_wires` transform already?